src/providers/xyz.js

```diff
@@ -43,8 +45,22 @@ XYZProvider.prototype = {
     d3.json(url, callback)
   },
 
-  invalidateCache: function () {
-    this.tileCache = {}
+  setReady: function () {
+    this.ready = true
+    this.layer.fire('ready')
```

Comment:
IMO, providers don't need to know about layers. Instead of introducing this new dependency (provider -> layer), I think providers can fire the `ready` event and then layers can do:

```
this.provider.on('ready', function () { ... })
```